src/python/WMCore/WMSpec/StdSpecs/StdBase.py

```diff
@@ -840,14 +840,16 @@ def factoryWorkloadConstruction(self, workloadName, arguments):
 
         Named this way so that nobody else will try to use this name.
         """
-        if 'OriginalRequestName' not in arguments:
-            self.masterValidation(schema=arguments)
+        if arguments.get("RequestType") == "Resubmission":
+            # It accepts both creation & assignment, thus skip master validation
             self.validateSchema(schema=arguments)
-        elif arguments.get("RequestType") == "Resubmission":
-            # Don't validate schema for cloned workflow
-            # But Resubmission workflow has to be validated
+        elif clone:
```

Comment:
should we check clone before checking the Resubmission? In case resubmission request is cloned.